cpp/fory/serialization/collection_serializer.h

```diff
@@ -385,21 +385,7 @@ struct Serializer<
       return Result<void, Error>();
     }
 
-    // Build header bitmap
-    bool has_null = false;
-    if constexpr (is_nullable_v<T>) {
```

Comment:
This is executed at compile time, if elememt is not nullable, the whole branch will be removed by compiler. There is not duplicate iteration.